README
NAME
    DB::Pluggable - Add plugin support for the Perl debugger

VERSION
    version 1.112001

SYNOPSIS
        $ cat ~/.perldb
        use DB::Pluggable;
        DB::Pluggable->run_with_config(\<<EOINI)
        [BreakOnTestNumber]

        [TypeAhead]
        type = {l
        type = c
        ifenv = DBTYPEAHEAD

        [StackTraceAsHTML]
        [DataPrinter]
        EOINI

    Then:

        $ perl -d foo.pl

DESCRIPTION
    This class adds plugin support to the Perl debugger. It is based on
    Brickyard, so see its documentation for details.

    You need to have a "~/.perldb" file (see perldebug for details) that
    invokes the plugin mechanism.

    Plugins should live in the "DB::Pluggable::Plugin::" namespace, like
    DB::Pluggable::Plugin::BreakOnTestNumber does.

METHODS
  run_with_config
    Convenience class method to create, initialize and run the plugin system
    with the given configuration file or scalar reference.

  plugins_with
    Like the method with the same name in Brickyard.

  init_from_config
    Like the method with the same name in Brickyard.

  enable_watchfunction
    Tells the debugger to call "DB::watchfunction()", which in turn calls
    the "watchfunction()" method of all plugins that consume the
    "-WatchFunction" role.

  run
    This method just calls the "initialize()" method of all plugins that
    consume the "-Initializer" role.

Plugin Phases
    This class is very much in beta, so it's more like a proof of concept.
    Therefore, not all roles - which more or less correspond to plugin
    phases - imaginable have been added, only the ones to make this demo
    work. If you want more roles or if the current roles don't work for you,
    let me know.

    The following roles exist:

    "-Initializer"
        See DB::Pluggable::Role::Initializer.

    "-WatchFunction"
        See DB::Pluggable::Role::WatchFunction.

    "-AfterInit"
        See DB::Pluggable::Role::AfterInit.

    "-Eval"
        See DB::Pluggable::Role::Eval. The debugger's "eval()" function is
        overridden so we can make it pluggable. Each plugin will get a
        chance to inspect the command line, which is the last line in
        $DB::evalarg and act on it. The plugin can return a code reference
        which will be executed after the original "DB::eval()" function has
        finished. Using the code reference you can undo any temporary
        changes you might have introduced to make your command work.

INSTALLATION
    See perlmodinstall for information and options on installing Perl
    modules.

BUGS AND LIMITATIONS
    No bugs have been reported.

    Please report any bugs or feature requests through the web interface at
    <http://rt.cpan.org/Public/Dist/Display.html?Name=DB-Pluggable>.

AVAILABILITY
    The latest version of this module is available from the Comprehensive
    Perl Archive Network (CPAN). Visit <http://www.perl.com/CPAN/> to find a
    CPAN site near you, or see <http://search.cpan.org/dist/DB-Pluggable/>.

    The development version lives at
    <https://github.com/garu/DB-Pluggable> and may be cloned from
    <git://github.com/garu/DB-Pluggable.git>. Instead of sending
    patches, please fork this project using the standard git and github
    infrastructure.

AUTHOR
    Marcel Gruenauer <marcel@cpan.org>

COPYRIGHT AND LICENSE
    The following copyright notice applies to all the files provided in this
    distribution, including binary files, unless explicitly noted otherwise.

    This software is copyright (c) 2008-2011 by Marcel Gruenauer.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Perl extension {{$dist->name}}

{{$NEXT}}

1.12  Wed, 08 Mar 2023 13:59:16 -0500
    - Update to work correctly with Data::Printer

1.112001  2011-07-19 21:02:33 Europe/Vienna
    - added VERSION numbers

1.112000  2011-07-19 20:30:08 Europe/Vienna
    - converted the distribution back to EU::MM style
    - doc fix for the Data::Printer plugin (thanks garu)

1.111751  2011-06-24 23:54:05 Europe/Vienna
    - doc fix: don't mention nonexistent roles. :)

1.111750  2011-06-24 23:04:34 Europe/Vienna
    - complete rewrite to move from Hook::Modular to Brickyard - it's more
      lightweight and easier to use

1.101051  2010-04-15 18:01:53 Europe/Vienna
    - fixed the db.eval hook handler

1.101050  2010-04-15 00:15:37 Europe/Vienna
    - added db.eval hook so you can define custom commands that take arguments
    - removed unused variables and code

1.100851  2010-03-26 20:56:57 Europe/Vienna
    - DB::Pluggable::Plugin now returns a true value

1.100850  2010-03-26 20:50:12 Europe/Vienna
    - converted the distribution to Dist::Zilla-style

0.04  Thu Jul 25 15:27:43 CEST 2008 (Marcel Gruenauer <marcel@cpan.org>)
    - Changed dist style and Changes back to standard. People didn't like it -
      the nail that sticks out gets hammered down.
    - Added standard test files; this will also help with CPANTS
    - Added DB::Pluggable::Plugin base class for plugins

0.03  Thu, 29 May 2008 12:04:18 -0000 (Marcel Gruenauer <marcel@cpan.org>)
    - set the version to 0.03
    - fixed first year, is 2008
    - .shipit: fixed svk.tagpattern

0.02  Thu, 17 Apr 2008 21:49:33 -0000 (Marcel Gruenauer <marcel@cpan.org>)
    - hiding the DB package from the PAUSE indexer
    - set version to 0.02

0.01  Thu, 17 Apr 2008 19:45:15 -0000 (Marcel Gruenauer <marcel@cpan.org>)
    - original version


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.030.
use strict;
use warnings;

use 5.010;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Add plugin support for the Perl debugger",
  "AUTHOR" => "Keith Carangelo <mail\@kcaran.com>, Marcel Gruenauer <marcel\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "DB-Pluggable",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010",
  "NAME" => "DB::Pluggable",
  "PREREQ_PM" => {
    "Brickyard" => "1.111750",
    "Brickyard::Accessor" => 0,
    "Brickyard::Role::Plugin" => 0,
    "Data::Printer" => 0,
    "Hook::LexWrap" => 0,
    "Role::Basic" => 0,
    "Test::Builder" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Carp" => 0,
    "File::Find" => 0,
    "File::Temp" => 0,
    "Scalar::Util" => 0,
    "Test::More" => "0.94"
  },
  "VERSION" => "1.12",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Brickyard" => "1.111750",
  "Brickyard::Accessor" => 0,
  "Brickyard::Role::Plugin" => 0,
  "Carp" => 0,
  "Data::Printer" => 0,
  "File::Find" => 0,
  "File::Temp" => 0,
  "Hook::LexWrap" => 0,
  "Role::Basic" => 0,
  "Scalar::Util" => 0,
  "Test::Builder" => 0,
  "Test::More" => "0.94",
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
